### PR TITLE
docs: update contributing documentation

### PR DESCRIPTION
**Pull request by bytemain**

### CODE REVIEW

File: .devcontainer/build.sh [hunk 1]
```diff
@@ -14,5 +14,5 @@ yarn workspace @affine/server-native build
 # Create database
 yarn workspace @affine/server prisma db push
 
-# Create user username: affine, password: affine
+# Create user username: affine, email: affine@affine.pro, password: affine
```
Comment (darkskygit) [inline .devcontainer/build.sh]: this script is outdated, server will init a dev@affine.pro account now

File: docs/developing-server.md [hunk 2]
```diff
@@ -90,6 +90,17 @@ yarn workspace @affine/server-native build
 yarn workspace @affine/native build
 ```
 
+## Optionally, create a default user
```
Comment (darkskygit) [inline docs/developing-server.md]: dont need this chapter, see first comment

File: .devcontainer/build.sh (hunk 1 above)
Comment (bytemain) [reply .devcontainer/build.sh]: so I should just delete this line and update the comments?

File: .devcontainer/build.sh [hunk 3]
```diff
@@ -13,6 +13,3 @@ yarn workspace @affine/server-native build
 
 # Create database
 yarn workspace @affine/server prisma db push
-
-# Create user username: affine, password: affine
-echo "INSERT INTO \"users\"(\"id\",\"name\",\"email\",\"email_verified\",\"created_at\",\"password\") VALUES('99f3ad04-7c9b-441e-a6db-79f73aa64db9','affine','affine@affine.pro','2024-02-26 15:54:16.974','2024-02-26 15:54:16.974+00','\$argon2id\$v=19\$m=19456,t=2,p=1\$esDS3QCHRH0Kmeh87YPm5Q\$9S+jf+xzw2Hicj6nkWltvaaaXX3dQIxAFwCfFa9o38A');" | yarn workspace @affine/server prisma db execute --stdin
```
Comment (bytemain) [inline .devcontainer/build.sh:18]: or should we do not remove these lines for compatible

File: .devcontainer/build.sh (hunk 1 above)
Comment (darkskygit) [reply .devcontainer/build.sh]: > so I should just delete this line and update the comments?

recommand add comments to doc

**Review by darkskygit — APPROVED** [review]

**Pull request merged**